src/Symfony/Component/Messenger/Stamp/ReceiverMessageIdStampInterface.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Stamp;
+
+/**
+ * Holds a message id, which is any value understood by a ListableReceiverInterface.
+ *
+ * @author Ryan Weaver <ryan@symfonycasts.com>
+ *
+ * @experimental in 4.3
+ */
+interface ReceiverMessageIdStampInterface extends StampInterface
```

Comment:
the sender also could have the same id - would it make sense to make it a MessageIdStamp?

Reply:
Hmm. The current use for this is specifically to be able to get this id and then use it in `ListableReceiverInterface::find($id)`. I think I get what you are saying: sometimes it's useful to know the "id" of a message from the sender too, even though that's not something we're using yet.

Indeed, I think that's not possible currently - it would need to be added to each sender specifically, I believe. I think it's outside of the scope. WDYT?

Reply:
Adding an interface means adding an abstraction. My point is: should we cover the "message has a transport id" with a single abstraction - or do we need two abstractions - one for receivers, one for senders? AFAIK, one id for the transport is a better description. The fact that senders don't implement the abstraction in another thing.

Reply:
Mmmmh - this makes me think: shouldn't we remove this interface and replace it by a dedicated stamp? It would make more sense to me.

Reply:
Hmmm. Would we set it on “send” so that it’s available for any middleware when it’s initially dispatched? And then on receive, we don’t set it? Unless it’s missing (because this message is only received but not sent from messenger)?

Reply:
Yes! or because the sender side of the transport doesn't report back the id - or it's not implemented yet.

Reply:
Done!